src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -13,6 +13,7 @@ internal partial class HtmlMarkupParser : TokenizerBackedParser<HtmlTokenizer>
     {
         private const string ScriptTagName = "script";
 
+        private static readonly SyntaxList<RazorSyntaxNode> EmptySyntaxList = new SyntaxListBuilder<RazorSyntaxNode>(0).ToList();
```

Comment:
If this is going to be commonly needed, then add a public readonly static field to `SyntaxList<>`. The compiler will ensure that only one instance is created by type.